DecipherNow/helm-charts
# enable mtls for control-api

**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

Enable TLS for gm-control-api
<br/>
<br/>
2. What **changes to custom.yaml** are required?

<br/>
<br/>

3. Have you documented any additional setup steps or configurations options?

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

To test, assuming you have minikube running:

- update requirements.yaml to point to the local charts for control and gm-control-api
- install greymatter:

 ```
helm dep up greymatter
helm install greymatter -f greymatter.yaml -f greymatter-secrets.yaml -f credentials.yaml --set global.environment=kubernetes -n gm-deploy
```
- look at logs for gm-control-api and see that tls was enabled
- check you can reach the gm-control-api through the edge
- delete the deployment
- in greymatter.yaml, set global.control_api_tls=false and global.services.controlApi.egressTLS=false
- repeat previous steps with tls false to make sure everything is working
<br/>
